pontis-ui/src/components/Navigation/CommandPallete.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from "react";
+import { useRouter } from "next/router";
+import classNames from "classnames";
+import { Dialog, Combobox, Transition } from "@headlessui/react";
+
+import { AssetKeys } from "../../hooks/oracle";
+import { ChevronRightIcon, SearchIcon } from "@heroicons/react/outline";
+import AssetCardName from "../Asset/AssetCardName";
+import { useSearch } from "../../providers/search";
+import { assetKeyToUrl } from "../../../utils/encodeUrl";
+
+interface CommandPallateProps {
+  isOpen: boolean;
+}
+
+const CommandPallate: React.FC<CommandPallateProps> = ({ isOpen }) => {
+  const router = useRouter();
+  const setSearch = useSearch();
+  const [query, setQuery] = useState("");
+
+  useEffect(() => {
+    /**
+     * Listens for shortcut to open command pallete.
+     * @param {KeyboardEvent} event keydown event
+     */
+    function onKeydown(event) {
+      if (event.key === "k" && (event.metaKey || event.ctrlKey)) {
+        // AFAIK (isOpen) => !isOpen instead of !isOpen allows us to remove isOpen from the dependency array.
+        // This prevents us from mounting and unmounting the event listeners on every render.
+        setSearch((isOpen) => !isOpen);
```

Comment:
How is passing this function to the setSearch hook that uses context get to the high level `isOpen` state set in the App component?

Reply:
Could you clarify your question here?